### PR TITLE
Ayelet Binder/webSocket count

### PR DESCRIPTION
**Pull request by ayeletBinder**

### CODE REVIEW

File: client-side/src/stories/header/Icon.jsx [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
-  );
-};
-
-export default MessageIcon;
\ No newline at end of file
+import React, { useState } from 'react';
+import { Badge, IconButton } from '@mui/material';
+import { MailOutline, Mail } from '@mui/icons-material';
+import useWebSocket from '../../webSocket';
+import { useSelector } from 'react-redux';
```
Comment (SariGlick) [inline client-side/src/stories/header/Icon.jsx:5]: reorder the imports

File: client-side/src/stories/header/Icon.jsx [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
+  const [openMesagges,setOpenMesagges] = useState(false);
+  const user = useSelector(state => state.user.currentUser);
+  const userId = user._id;
+  const { cntUnreadMessages } = useWebSocket(userId);
+
+  const hasUnreadMessages = cntUnreadMessages > 0;
+
+
+  return (
+    <IconButton onClick={()=>{setOpenMesagges(!openMesagges);}} className="iconButton">
```
Comment (SariGlick) [inline client-side/src/stories/header/Icon.jsx:19]: remove ;

File: server-side/.env [hunk 3]
```diff
@@ -1,6 +1,5 @@
-
-PORT=3000
+PORT=5000
 password=EuACQK5n92JI3vn4
-DB_URL="mongodb://127.0.0.1:27017/TimeOut"
+DB_URL=mongodb+srv://sh3000444:EuACQK5n92JI3vn4@cluster0.2ttcjcg.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0
```
Comment (SariGlick) [inline server-side/.env:3]: return the previous value

**Review by Shoshana-Yudelevitz — COMMENTED** [review]
Ayelet Binder - webSocket count

**Review by SariGlick — CHANGES_REQUESTED** [review]

File: client-side/package.json [hunk 4]
```diff
@@ -65,8 +67,6 @@
     ]
   },
   "devDependencies": {
-    "@babel/preset-env": "^7.25.4",
-    "@babel/preset-react": "^7.24.7",
```
[the remark this answers is not in the document]
Comment (ayeletBinder) [reply client-side/package.json]: i did it

File: client-side/package.json [hunk 5]
```diff
@@ -80,35 +80,19 @@
     "@storybook/react": "^8.1.11",
     "@storybook/react-webpack5": "^8.1.11",
     "@storybook/test": "^8.1.11",
-    "babel-jest": "^29.7.0",
```
[the remark this answers is not in the document]
Comment (ayeletBinder) [reply client-side/package.json:83]: i did it

File: client-side/package.json [hunk 6]
```diff
@@ -80,35 +80,19 @@
...
     "eslint-config-react-app": "^7.0.1",
     "eslint-plugin-react": "^7.27.1",
     "eslint-plugin-storybook": "^0.8.0",
     "globals": "^15.8.0",
-    "identity-obj-proxy": "^3.0.0",
```
[the remark this answers is not in the document]
Comment (ayeletBinder) [reply client-side/package.json:88]: I did it

File: client-side/package.json [hunk 7]
```diff
@@ -80,35 +80,19 @@
...
+    "install": "^0.13.0",
+    "jest": "^29.7.0",
     "jest-environment-jsdom": "^29.7.0",
-    "jest-transform-stub": "^2.0.0",
```
[the remark this answers is not in the document]
Comment (ayeletBinder) [reply client-side/package.json:90]: I did it

File: client-side/package.json [hunk 8]
```diff
@@ -80,35 +80,19 @@
...
     "notistack": "^3.0.1",
+    "npm": "^10.8.2",
     "prop-types": "^15.8.1",
+    "react-apollo": "^3.1.5",
     "react-scripts": "^5.0.1",
-    "sass": "^1.78.0",
+    "sass": "^1.77.6",
     "storybook": "^8.1.11"
-  },
-  "jest": {
```
[the remark this answers is not in the document]
Comment (ayeletBinder) [reply client-side/package.json:97]: I did it

File: client-side/src/stories/header/Icon.jsx (hunk 2 above)
Comment (ayeletBinder) [reply client-side/src/stories/header/Icon.jsx]: i did it

File: client-side/src/stories/header/Icon.jsx [hunk 9]
```diff
@@ -1,27 +1,29 @@
...
+const MessageIcon = () => {
+  const user = useSelector(state => state.user.currentUser);
+  const userId = user._id;
+  const { cntUnreadMessages } = useWebSocket(userId);
+
+  const hasUnreadMessages = cntUnreadMessages > 0;
+
+
+  return (
+    <IconButton className="iconButton">
```
Comment (SariGlick) [inline client-side/src/stories/header/Icon.jsx:17]: Did you mean to remove the management of the open/close the messages component? (I meant only to remove the ; of the onClick, not the whole click event)

**Review by SariGlick — CHANGES_REQUESTED** [review]

File: server-side/.env (hunk 3 above)
Comment (ayeletBinder) [reply server-side/.env]: i did it

File: client-side/src/stories/header/Icon.jsx (hunk 1 above)
Comment (ayeletBinder) [reply client-side/src/stories/header/Icon.jsx]: i did it

File: client-side/package.json [hunk 10]
```diff
@@ -11,36 +11,38 @@
...
     "redux": "^5.0.1",
     "styled-components": "^6.1.11",
     "web-vitals": "^2.1.4",
+    "ws": "^8.18.0",
     "yup": "^1.4.0"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "jest --verbose",
```
[the remark this answers is not in the document]
Comment (ayeletBinder) [reply client-side/package.json]: i did it

File: client-side/src/stories/header/Icon.jsx (hunk 9 above)
Comment (ayeletBinder) [reply client-side/src/stories/header/Icon.jsx]: ok, actually i didnt understand why to remove it...

**Review by EA171 — APPROVED** [review]

**Pull request merged**